app/controllers/api/service_templates_controller.rb

```diff
@@ -4,5 +4,12 @@ class ServiceTemplatesController < BaseController
     include Subcollections::Tags
     include Subcollections::ResourceActions
     include Subcollections::ServiceRequests
+
+    def create_resource(type, _id, data)
+      catalog_item_type = data['type'].try(:constantize) || collection_class(type)
```

Comment:
we really shouldn't be running constantize on user specified data.  Prefer if model returned the class for us.  Are these known classes, or subclasses of known base classes ?  

see app/models/miq_request's class_from_request_data() that we call from API.  Might need something similar here.